src/ring_theory/polynomial/cyclotomic/expand.lean

```diff
@@ -124,7 +124,8 @@ section char_p
 /-- If `R` is of characteristic `p` and `¬p ∣ n`, then
 `cyclotomic (n * p) R = (cyclotomic n R) ^ (p - 1)`. -/
 lemma cyclotomic_mul_prime_eq_pow_of_not_dvd (R : Type*) {p n : ℕ} [hp : fact (nat.prime p)]
-  [ring R] [char_p R p] (hn : ¬p ∣ n) : cyclotomic (n * p) R = (cyclotomic n R) ^ (p - 1) :=
+  [ring R] [algebra (zmod p) R] (hn : ¬p ∣ n) [algebra (zmod p) R]:
```

Comment:
Same here: IMO the rule should be "use `char_p` unless the _statement_ needs the algebra instance.

Reply:
Mmm, I don't have an opinion of this, both solutions are fine.